test/unit/app-lifecycle.spec.js

```diff
@@ -99,6 +99,38 @@ describe('App-Lifecycle', function() {
     });
   });
 
+  describe('when restarting the app', function() {
+    beforeEach(function() {
+      this.myApp.start({
+        state: { 'foo': 'bar' }
+      });
+    });
+
+    it('should be stopped', function() {
+      this.myApp.restart();
+
+      expect(this.stopStub).to.have.been.calledOnce;
+    });
+
+    it('should trigger start event twice', function() {
+      expect(this.startStub).to.have.been.calledOnce;
+
+      this.myApp.restart();
+
+      expect(this.startStub).to.have.been.calledTwice;
+    });
+
+    it('should maintain the previous app\'s state', function() {
```

Comment:
super nit-picky but "app's previous state" is a bit less ambiguous as to what "previous" is referring to.